### ISSUE TITLE
Eclipse (both Java and Jee) crashes when just opening Help -> Eclipse Marketplace

### ISSUE THREAD

**Issue by mjl269** [eclipse-platform/eclipse.platform.swt#1185]
**Describe the bug**
Eclipse was just installed from eclipse-java-2024-03-R-linux-gtk-x86_64.tar.gz on Ubuntu 22.04 LTS with GTK3 and GTK4 installed, and still has no projects.

Just opening Help -> Eclipse Marketplace it crashes.

The main error messages (see appended eclCrash.log) are:
Gtk-CRITICAL **: 18:15:57.794: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar
Gdk-CRITICAL **: 18:18:32.607: gdk_window_create_gl_context: assertion 'GDK_IS_WINDOW (window)' failed

[hs_err_pid6635.log](https://github.com/eclipse-platform/eclipse.platform.swt/files/15030133/hs_err_pid6635.log)
[eclCrash.log](https://github.com/eclipse-platform/eclipse.platform.swt/files/15030135/eclCrash.log)
[lsb_release.log](https://github.com/eclipse-platform/eclipse.platform.swt/files/15030136/lsb_release.log)

**To Reproduce**
After Eclipse start just try Help -> Eclipse Marketplace. After loading is complete, and before any item is displayed, the crash occurs.

**Expected behavior**
The crash shouldn't occur.

**Screenshots**
n/a

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [ ] All OS
- - [ ] Windows
- - [x] Linux
- - [ ] macOS 

2. Additional OS info (e.g. OS version, Linux Desktop, etc)

3. JRE/JDK version
 OpenJDK 64-Bit Server VM Temurin-17.0.10+7 (17.0.10+7, mixed mode, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)

**Version since**
 Eclipse 2024-03

**Workaround (or) Additional context**
~No workaround possible.~
Quote from https://github.com/eclipse-platform/eclipse.platform.swt/issues/1185#issuecomment-2105098212:
> i have the same problem "export WEBKIT_DISABLE_DMABUF_RENDERER=1" fixed it. Ubuntu 24.04 lts and Eclipse 2024-03

Crashes also occur during normal Eclipse usage at unpredictable  time, but shortly after starting.

**Comment by Harsh-Noise** [comment]
Sounds like the same bug I was facing. Does it also crash when you type ctrl+space for code autocomplete?
If so, it's a problem upstream with Webkit. You need to disable WebkitGtk hardware-acceleration to stop the crashing. Do this by [setting the environment variable](https://www.serverlab.ca/tutorials/linux/administration-linux/how-to-set-environment-variables-in-linux/) WEBKIT_DISABLE_DMABUF_RENDERER=1. Make sure to read to the end and make it persist after a reboot.

**Comment by billykd4dpb** [comment]
How do i set the variable sence linux mint dose not have wayland it cant be that i looked at the info on how to do it didnt understand thanks

**Comment by rober81** [comment]
i have the same problem  "export WEBKIT_DISABLE_DMABUF_RENDERER=1" fixed it. Ubuntu 24.04 lts and Eclipse 2024-03

**Comment by fedejeanne** [comment]
@billykd4dpb does exporting the variable fix the issue? If yes, can you please update the description of this bug and add the workaround?

**Comment by dvlierop** [comment]
> @billykd4dpb does exporting the variable fix the issue? If yes, can you please update the description of this bug and add the workaround?

Yes, it does (Eclipse 2023-12 on Ubuntu 24.04.1 LTS); I've added ```WEBKIT_DISABLE_DMABUF_RENDERER=1``` to /etc/environment

**Comment by opedrorendeiro** [comment]
> Sounds like the same bug I was facing. Does it also crash when you type ctrl+space for code autocomplete? If so, it's a problem upstream with Webkit. You need to disable WebkitGtk hardware-acceleration to stop the crashing. Do this by [setting the environment variable](https://www.serverlab.ca/tutorials/linux/administration-linux/how-to-set-environment-variables-in-linux/) WEBKIT_DISABLE_DMABUF_RENDERER=1. Make sure to read to the end and make it persist after a reboot.

It worked for me (Eclipse 2022-12 with Ubuntu 22.04.4 LTS). Thank you so much!!

**Comment by akurtakov** [comment]
Do someone still reproduce this one with latest release?

**Comment by akurtakov** [comment]
I'm closing this one as I can't reproduce and without someone actively testing and being able to reproduce the issue it is non actionable.
